services/src/main/java/org/keycloak/email/DefaultEmailSenderProvider.java

```diff
@@ -185,16 +188,29 @@ protected String retrieveEmailAddress(UserModel user) {
     }
 
     private void setupTruststore(Properties props) {
-        JSSETruststoreConfigurator configurator = new JSSETruststoreConfigurator(session);
-
-        SSLSocketFactory factory = configurator.getSSLSocketFactory();
-        if (factory != null) {
-            props.put("mail.smtp.ssl.socketFactory", factory);
-            if (configurator.getProvider().getPolicy() == HostnameVerificationPolicy.ANY) {
-                props.setProperty("mail.smtp.ssl.trust", "*");
-                props.put("mail.smtp.ssl.checkserveridentity", Boolean.FALSE.toString());
-            }
-        }
+         JSSETruststoreConfigurator configurator = new JSSETruststoreConfigurator(session);
+         if (configurator.getProvider().getPolicy() == HostnameVerificationPolicy.ANY) {
```

Comment:
It would make sense to handle this as part of the logic in the FileTruststoreProvider / JSSETruststoreConfigurator, so that it returns a SSLSocketFactory that is already setup for ANY rather than making callers do this.

Reply:
(y)

Reply:
Done